### ISSUE TITLE
mongodb_status fails on Arbiter because of is_auth_enabled check

### ISSUE THREAD

**Issue by RayJin2000** [ansible-collections/community.mongodb#651]
##### SUMMARY

If you try to get the Replication Status of Arbiter with no active authentication you get the following error:
```paste below
Unable to determine if auth is enabled: Traceback (most recent call last):
    pymongo.errors.NotPrimaryError: node is not in primary or recovering state, full error: {'topologyVersion': {'processId': ObjectId('6655f606ee98c67bac861bed'), 'counter': 1}, 'ok': 0.0, 'errmsg': 'node is not in primary or recovering state', 'code': 13436, 'codeName': 'NotPrimaryOrSecondary'}
```
The Problem seems to be here: https://github.com/ansible-collections/community.mongodb/blob/master/plugins/module_utils/mongodb_common.py#L294
When you try to execute db.adminCommand('listDatabases', 1.0) on an Arbiter ( with no authentication ) you get the error NotPrimaryOrSecondary([13436](https://www.mongodb.com/docs/manual/reference/error-codes/#mongodb-error-13436) ) and not as expected Unauthorized([13](https://www.mongodb.com/docs/manual/reference/error-codes/#mongodb-error-13))

This looks similar to this Bug: https://github.com/ansible-collections/community.mongodb/issues/136
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
mongodb_status 

##### ANSIBLE VERSION

```paste below
ansible [core 2.16.6]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.12/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.12.3 (main, Apr 17 2024, 00:00:00) [GCC 14.0.1 20240411 (Red Hat 14.0.1-0)] (/usr/bin/python3)
  jinja version = 3.1.4
  libyaml = True

```

##### COLLECTION VERSION

```paste below
 ansible-galaxy collection list community.mongodb

# /usr/lib/python3.12/site-packages/ansible_collections
Collection        Version
----------------- -------
community.mongodb 1.7.3  
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

MongoDB 7.0.11 on RHEL9

##### STEPS TO REPRODUCE

```yaml
    - name: Get replica_set status
      community.mongodb.mongodb_status:
        login_host: "{{ inventory_hostname }}"
        login_port: "{{ mongodb_port }}"
        login_user: "{{ mongodb_admin_user }}"
        login_password: "{{ mongodb_admin_default_pw }}"
        replica_set: "{{ mongodb_replset_name }}"
        auth_mechanism: "SCRAM-SHA-256"
        poll: 6
        interval: 10
```

##### EXPECTED RESULTS

```paste below
  mongodb_status:
    changed: false
    failed: false
    failed_when_result: false
    iterations: 1
    msg: replicaset is in a converged state
    replicaset:
     mongodb11-test:27015: PRIMARY
     mongodb21-test:27015: SECONDARY
     mongodb31-test:27015: ARBITER

```

##### ACTUAL RESULTS

```paste below
Unable to determine if auth is enabled: Traceback (most recent call last):
      File "/tmp/ansible_community.mongodb.mongodb_status_payload_tqnr2bm4/ansible_community.mongodb.mongodb_status_payload.zip/ansible_collections/community/mongodb/plugins/module_utils/mongodb_common.py", line 294, in is_auth_enabled
        myclient['admin'].command('listDatabases', 1.0)
      File "/opt/dw_mongo_virtualenv/lib64/python3.11/site-packages/pymongo/_csot.py", line 107, in csot_wrapper
        return func(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/dw_mongo_virtualenv/lib64/python3.11/site-packages/pymongo/database.py", line 894, in command
        return self._command(
               ^^^^^^^^^^^^^^
      File "/opt/dw_mongo_virtualenv/lib64/python3.11/site-packages/pymongo/database.py", line 743, in _command
        return conn.command(
               ^^^^^^^^^^^^^
      File "/opt/dw_mongo_virtualenv/lib64/python3.11/site-packages/pymongo/helpers.py", line 322, in inner
        return func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/dw_mongo_virtualenv/lib64/python3.11/site-packages/pymongo/pool.py", line 968, in command
        return command(
               ^^^^^^^^
      File "/opt/dw_mongo_virtualenv/lib64/python3.11/site-packages/pymongo/network.py", line 192, in command
        helpers._check_command_response(
      File "/opt/dw_mongo_virtualenv/lib64/python3.11/site-packages/pymongo/helpers.py", line 217, in _check_command_response
        raise NotPrimaryError(errmsg, response)
    pymongo.errors.NotPrimaryError: node is not in primary or recovering state, full error: {'topologyVersion': {'processId': ObjectId('6655f606ee98c67bac861bed'), 'counter': 1}, 'ok': 0.0, 'errmsg': 'node is not in primary or recovering state', 'code': 13436, 'codeName': 'NotPrimaryOrSecondary'}
```

**Comment by rhysmeister** [comment]
Not able to replicate this. test case in https://github.com/ansible-collections/community.mongodb/pull/657

This is probably just a timing issue. I could perhaps add a retry on certain conditions. So if NotPrimaryOrSecondary is thrown, I could pause for a bit and retr.

**Comment by RayJin2000** [comment]
Hi, I looked at your tests and you only test replica sets with full members, but the error only occurs on an arbiter.

**Comment by rhysmeister** [comment]
The test with an Arbiter is here...

https://github.com/ansible-collections/community.mongodb/pull/657/files

"i believe you have to activate Authentication" Above you state "Arbiter with no active authentication" - What do you mean exactly?

**Comment by RayJin2000** [comment]
I only looked at the 150 test runs and found nothing suitable at mongodb_status. Sorry, my memory is bad. I'll try to reproduce the error again.

**Comment by RayJin2000** [comment]
I found our problematic setup Again. We have 3 Mongos without Authentication: Primary, Secondary & Arbiter. You need to ask the Arbiter direct to get the error.

**Comment by rhysmeister** [comment]
@RayJin2000 Replicated the issue and fixed it... https://github.com/ansible-collections/community.mongodb/pull/657

Tests use a 3 member cluster, 2 standard and 1 arbiter node, with and without auth.

**Comment by rhysmeister** [comment]
Released in 1.7.5.
